### PR TITLE
OPS-2863 Update node version to 16

### PR DESCRIPTION
**Pull request by MerylWang**
- [x] bring fork (physion/ec2-github-runner) up to date with original (machulav/ec2-github/runner)
- [x] update node version to 16
- [x] ~~build new dist `npm run package` or `yarn package`~~

### CODE REVIEW

File: src/aws.js [hunk 1]
```diff
@@ -2,6 +2,32 @@ const AWS = require('aws-sdk');
...
+      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
+      'export RUNNER_ALLOW_RUNASROOT=1',
+      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
+      './run.sh',
+    ];
+  }
+}
+
 async function startEc2Instance(label, githubRegistrationToken) {
   const ec2 = new AWS.EC2();
```
Comment (MerylWang) [inline src/aws.js:43]: place 1 of merge conflict merging with upstream: 
the original wanted to use  `const userData = buildUserDataScript(...)` 
I kept the current userData

File: src/config.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ class Config {
       label: core.getInput('label'),
       ec2InstanceId: core.getInput('ec2-instance-id'),
       iamRoleName: core.getInput('iam-role-name'),
-      keyPairName: core.getInput('key-pair-name')
+      keyPairName: core.getInput('key-pair-name'),
```
Comment (MerylWang) [inline src/config.js:16]: place 2 of merge conflict: 
original wanted to add `runnerHomeDir` and remove `keyPairName` I think. I kept both.

File: action.yml [hunk 3]
```diff
@@ -78,5 +82,5 @@ outputs:
       EC2 Instance Id of the created runner.
       The id is used to terminate the EC2 instance when the runner is not needed anymore.
 runs:
-  using: node12
+  using: node16
```
Comment (MerylWang) [inline action.yml:85]: our runner image already has node 16 installed
https://github.com/actions/runner-images/blob/ubuntu20/20221018.2/images/linux/Ubuntu2004-Readme.md

**Comment by MerylWang** [comment]
test node version got bumped successfully by running a GitHub workflow through this branch?
if the version is bumped, GitHub shouldn't complain about the deprecation of this action

Update: Test ran successfully [here](https://github.com/physion/ovation.io/actions/runs/3325459462/jobs/5498166348)

File: src/config.js (hunk 2 above)
Comment (johnsonrw82) [reply src/config.js:16]: you've got a trailing comma after the last entry in the list

File: src/aws.js (hunk 1 above)
Comment (johnsonrw82) [reply src/aws.js:43]: The function should be fine to switch to, we just need to make sure the user data contents have what we need.

I would use this resolution:
```
// User data scripts are run as the root user
function buildUserDataScript(githubRegistrationToken, label) {
  if (config.input.runnerHomeDir) {
    // If runner home directory is specified, we expect the actions-runner software (and dependencies)
    // to be pre-installed in the AMI, so we simply cd into that directory and then start the runner
    return [
      '#!/bin/bash',
      `cd "${config.input.runnerHomeDir}"`,
      'export RUNNER_ALLOW_RUNASROOT=1',
      'sudo echo "export RUNNER_ALLOW_RUNASROOT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false" >> /etc/profile.d/env.sh',
      `./config.sh --unattended --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
      './run.sh',
    ];
  } else {
    return [
      '#!/bin/bash',
      'mkdir actions-runner && cd actions-runner',
      'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
      'curl -O -L https://github.com/actions/runner/releases/download/v2.286.0/actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
      'export RUNNER_ALLOW_RUNASROOT=1',
      'sudo echo "export RUNNER_ALLOW_RUNASROOT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false" >> /etc/profile.d/env.sh',
      `./config.sh --unattended --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
      './run.sh',
    ];
  }
}
```

File: src/config.js (hunk 2 above)
Comment (MerylWang) [reply src/config.js:16]: removed

File: src/aws.js (hunk 1 above)
Comment (MerylWang) [reply src/aws.js:43]: switched

**Review by johnsonrw82 — APPROVED** [review]

File: src/config.js (hunk 2 above)
Comment (MerylWang) [reply src/config.js:16]: actually I think `runnerHomeDir` can probably go because it's used by the class

File: src/aws.js (hunk 1 above)
Comment (MerylWang) [reply src/aws.js:43]: `config` doesn't set `runnerHomeDir` right now so I think it's always going to be the `else` clause

**Pull request merged**